gcc/rust/checks/errors/rust-feature-gate.cc

```diff
@@ -17,11 +17,19 @@
 // <http://www.gnu.org/licenses/>.
 
 #include "rust-feature-gate.h"
+#include "config.h"
+#include "is-a.h"
 #include "rust-abi.h"
+#include "rust-ast.h"
 #include "rust-attribute-values.h"
 #include "rust-ast-visitor.h"
 #include "rust-feature.h"
 #include "rust-ast-full.h"
+#include "rust-item.h"
+#include "rust-path.h"
+#include "rust-type.h"
+#include "rust-tyty.h"
+#include <iostream>
```

Comment:
this should use rust-system.h i dont see why you need iostream and rust-tyty.h for the type abstractions